querqy-for-lucene/querqy-lucene/src/main/java/querqy/lucene/contrib/rewrite/wordbreak/WordBreakCompoundRewriter.java

```diff
@@ -54,12 +57,14 @@
      * @param reverseCompoundTriggerWords Query tokens found as keys in this map will trigger the creation of a reverse compound of the surrounding tokens.
      * @param maxDecompoundExpansions The maximum number of decompounds to add to the query
      * @param verifyDecompoundCollation Iff true, verify that all parts of the compound cooccur in dictionaryField after decompounding
+     * @param protectedWords The "false-positive" set of terms that should never be split or be result of a combination
      */
     public WordBreakCompoundRewriter(final LuceneWordBreaker wordBreaker, final LuceneCompounder compounder,
                                      final IndexReader indexReader,
                                      final boolean lowerCaseInput, final boolean alwaysAddReverseCompounds,
                                      final TrieMap<Boolean> reverseCompoundTriggerWords,
-                                     final int maxDecompoundExpansions, final boolean verifyDecompoundCollation) {
+                                     final int maxDecompoundExpansions, final boolean verifyDecompoundCollation,
+                                     final CharArraySet protectedWords) {
```

Comment:
We should consistently use either TrieMap<Boolean> or CharArraySet for a 'fast set of character sequences' (compare reverseCompoundTriggerWords vs CharArraySet). I think that TrieMap would be a tiny bit faster as it doesn't calculate a hashcode, unlike CharArraySet.

Reply:
Have changed CharArraySet -> TrieMap